### PR TITLE
Switch to file header in spring-cloud-build

### LINKED ISSUE #1245: Checkstyle: remove checkstyle-header.txt in favor of one provided in the spring-cloud-build
After fixing https://github.com/spring-cloud/spring-cloud-build/issues/62 we don't need our local `checkstyle-header.txt` any more.

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #1245.

### CODE REVIEW

**Comment by meltsufin** [comment]
@artembilan PTAL, shouldn't take long :laughing:

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java [hunk 1]
```diff
@@ -1,17 +1,17 @@
 /*
- *  Copyright 2017 original author or authors.
+ * Copyright 2017-2017 the original author or authors.
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java]: There many of these single year ranges. Is that correct?

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/AppEngineCondition.java [hunk 2]
```diff
@@ -1,17 +1,17 @@
 /*
- *  Copyright 2017 original author or authors.
+ * Copyright 2017-2017 the original author or authors.
```
[the remark this answers is not in the document]
Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/AppEngineCondition.java]: So, we should have 2017-2018 everywhere, right?

[the remark this answers is not in the document]
Comment (artembilan) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/AppEngineCondition.java]: That's correct. Even if it is new class, the first year is about the date we started project.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/DatastoreRepositoriesAutoConfiguration.java [hunk 3]
```diff
@@ -1,17 +1,17 @@
 /*
- *  Copyright 2018 original author or authors.
+ * Copyright 2018-2018 the original author or authors.
```
[the remark this answers is not in the document]
Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/DatastoreRepositoriesAutoConfiguration.java]: Updated everywhere.

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java (hunk 1 above)
Comment (meltsufin) [reply spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java]: Yes, but see Artem's comments. We're changing it to 2017-2018.
Are you requesting changes to the PR?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java]: Would just "2017" be acceptable?

Comment (meltsufin) [reply spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java]: Not according to Spring Boot header style.

Comment (artembilan) [reply spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubProducerDestination.java]: No, see related discussion in the issue for this PR.

**Pull request merged**